src/Connectors/PhpRedisConnector.php

```diff
@@ -0,0 +1,260 @@
+
+        'sentinel_persistent' => null,
+        'sentinel_read_timeout' => null,
+    ];
+
+    /**
+     * Create a new Redis Sentinel connection from the provided configuration
+     * options
+     *
+     * @param array $server  The client configuration for the connection
+     * @param array $options The global client options shared by all Sentinel
+     * connections
+     *
+     * @return PhpRedisConnection The Sentinel connection containing a configured
+     * PhpRedis Client
+     */
+    public function connect(array $servers, array $options = [ ])
+    {
+        // Set the initial Sentinel servers.
+        $this->servers = array_map(function ($server) {
+            return $this->formatServer($server);
+        }, $servers);
+
+        // Set the connector options.
+        $this->retryLimit = (int) (isset($options['connector_retry_limit']) ? $options['connector_retry_limit'] : 20);
+        $this->retryWait = (int) (isset($options['connector_retry_wait']) ? $options['connector_retry_wait'] : 1000);
+
+        // Merge the global options shared by all Sentinel connections with
+        // connection-specific options
+        $clientOpts = array_merge($options, Arr::pull($servers, 'options', [ ]));
+
+        // Extract the array of Sentinel connection options from the rest of
+        // the client options
+        $sentinelOpts = array_intersect_key($clientOpts, $this->sentinelKeys);
+
+        // Filter the Sentinel connection options elements from the client
+        // options array
+        $clientOpts = array_diff_key($clientOpts, $this->sentinelKeys);
+
+        // Create a client by calling the Sentinel servers
+        $connector = function () use ($options) {
+            return $this->createClientWithSentinel($options);
+        };
+
+        // Create a connection and retry if this fails.
+        $connection = $this->retryOnFailure(function () use ($connector) {
+            return $connector();
+        });
+
+        return new PhpRedisConnection($connection, $connector, $sentinelOpts);
+    }
+
+    /**
+     * Create the Redis client instance
+     *
+     * @param  array  $options
+     * @return Redis
+     *
+     * @throws LogicException
+     */
+    protected function createClientWithSentinel(array $options)
+    {
+        $servers = $this->servers;
+        $service = isset($options['service']) ? $options['service'] : 'mymaster';
+        $timeout = isset($options['sentinel_timeout']) ? $options['sentinel_timeout'] : 0;
+        $persistent = isset($options['sentinel_peristent']) ? $options['sentinel_peristent'] : null;
+        $retryWait = isset($options['retry_wait']) ? $options['retry_wait'] : 0;
+        $readTimeout = isset($options['sentinel_read_timeout']) ? $options['sentinel_read_timeout'] : 0;
+        $parameters = isset($options['parameters']) ? $options['parameters'] : [];
+
+        // Shuffle the servers to perform some loadbalancing.
+        shuffle($servers);
```

Comment:
Seems like the most trivial solution, but it's [probably not suitable for actual load balancing](https://3v4l.org/vMitj).

Reply:
True, but i don't think we should implement full on loadbalancing, there are other tools for that.

Reply:
It's not that I dislike the trivial solution of using a random server to connect to. What I was trying to point out is that `shuffle` will most likely not distribute the load evenly because the used random number generator is quite bad. If you run the linked snippet a few times, it will be clear what I mean.

Maybe this isn't an issue after all though, since connecting to a Sentinel cluster often happens through a load balancer (e.g. Kubernetes Service) anyway.